packages/mdc-tab-scroller/foundation.js

```diff
@@ -0,0 +1,400 @@
+/**
+  * @license
+  * Copyright 2018 Google Inc. All Rights Reserved.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License")
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+*/
+
+import MDCFoundation from '@material/base/foundation';
+import {cssClasses, strings} from './constants';
+/* eslint-disable no-unused-vars */
+import {MDCTabScrollerAnimation, MDCTabScrollerEdges, MDCTabScrollerAdapter} from './adapter';
+import MDCTabScrollerRTL from './rtl-scroller';
+/* eslint-enable no-unused-vars */
+import MDCTabScrollerRTLDefault from './rtl-default-scroller';
+import MDCTabScrollerRTLNegative from './rtl-negative-scroller';
+import MDCTabScrollerRTLReverse from './rtl-reverse-scroller';
+
+const INTERACTION_EVENTS = ['wheel', 'touchstart', 'pointerdown', 'mousedown', 'keydown'];
```

Comment:
I'm assuming we want to handle mousedown and keydown for the purpose of if a user selects a tab while scroll is happening? But what about wheel?

Reply:
Wheel, touchstart, pointerdown, mousedown, and keydown are all used to stop the animation if someone interacts with the scroller while it's animating. That's to simulate the normal scroll behavior where if you click on a page while the page is doing a native scroll animation, the scroll animation stops.